projects/cm_mcu/CommandLineTask.c

```diff
@@ -848,6 +851,72 @@ static int execute(void *p, int argc, char **argv)
   return 0;
 }
 
+static BaseType_t first_mcu_ctl(int argc, char **argv, char *m)
+{
+
+  if (read_eeprom_single(EEPROM_ID_SN_ADDR) == 0xffffffff) {
+    uint32_t board_id, rev, ps_mask, data;
+    uint64_t pass, addr_id, addr_ff, addr_ps;
+    board_id = strtoul(argv[1], NULL, 16);
+    rev = strtoul(argv[2], NULL, 16);
+    ff_USER_mask = strtoul(argv[3], NULL, 16);
+    ps_mask = strtoul(argv[4], NULL, 16);
+    snprintf(m, SCRATCH_SIZE, "Registering board_id %lx revision %lx, USER ff mass %lx and PS ignore mask %lx \r\n", board_id, rev, ff_USER_mask, ps_mask);
+
+    pass = 0x12345678;
+    addr_id = 0x40; // internal eeprom block for board id
+    addr_ff = 0x44; // internal eeprom block for ps ignore fail
+    addr_ps = 0x48; // internal eeprom block for ps ignore fail
+
+    data = (board_id << 16) + rev;
+    uint64_t block = EEPROMBlockFromAddr(addr_id);
+
+    uint64_t unlock = EPRMMessage((uint64_t)EPRM_UNLOCK_BLOCK, block, pass);
+    xQueueSendToBack(xEPRMQueue_in, &unlock, portMAX_DELAY);
+
+    uint64_t message = EPRMMessage((uint64_t)EPRM_WRITE_SINGLE, addr_id, data);
+    xQueueSendToBack(xEPRMQueue_in, &message, portMAX_DELAY);
+
+    uint64_t lock = EPRMMessage((uint64_t)EPRM_LOCK_BLOCK, block << 32, 0);
+    xQueueSendToBack(xEPRMQueue_in, &lock, portMAX_DELAY);
+
+    data = ff_USER_mask;
+    block = EEPROMBlockFromAddr(addr_ff);
+
+    unlock = EPRMMessage((uint64_t)EPRM_UNLOCK_BLOCK, block, pass);
```

Comment:
I think you only need to do the unlock/lock sequence 1x right? No need to repeatedly unlock/lock the EEPROM.

Reply:
Yep, you're right. I thought that it lock/unlock per block address. Done.